File: RustSideStory/SampleSource/iflet.swift
// if let introduced with Swift

let x: String? = "Hello World"
let nothing: Int? = nil

if let x = x {
	print("x found! \(x)")
}
if case .some(let x) = x {
	print("x found! \(x)")
}
if let nothing = nothing {
	print("nothing found! \(nothing)")
}

switch x {
	case .some(let x): print("x found! \(x)")
	default: break
}

File: RustSideStory/SampleSource/iflet.rs
// if let introduced with RFC 160

let x = Some("Hello World");
let nothing: Option<i32> = None;

if let Some(x) = x {
	println!("x found! {}", x);
}
if let Some(nothing) = nothing {
	println!("nothing found! {}", nothing);
}

match x {
	Some(x) => println!("x found! {}", x),
	_ => {},
}
